app/models/miq_worker.rb

```diff
@@ -455,6 +455,18 @@ def clean_active_messages
     end
   end
 
+  def error_out_tasks_with_active_queue_message
+    message = "Task Handler: [#{friendly_name}] ID [#{id}] has been deleted!"
+    processed_messages.includes(:miq_task).each do |m|
+      task = m.miq_task
+      next unless task
+
+      # Note, this is done synchronously from destroy because workers have 1 message they're currently "handling"
+      # and each message can only have 1 task, so this should be very fast even for many workers.
+      task.update_status(MiqTask::STATE_FINISHED, MiqTask::STATUS_ERROR, message)
+    end
```

Comment:
Is it something as simple as a single query?
Guess that does not log task updates
And not sure about other logic. so maybe the n+1 is better.

```suggestion
    MiqTask.where(:id => processed_messages.select(:miq_task_id)).update_all(
      :status => MiqTask::STATUS_ERROR,
      :message => message,
      :state => MiqTask::STATE_FINISHED
    )
```